pymc3_hmm/step_methods.py

```diff
@@ -156,14 +194,30 @@ def astep(self, point, log_lik_fn, inputs):
         M = gamma_0.shape[-1]
         N = point.shape[-1]
 
-        # TODO: Why won't broadcasting work with `log_lik_fn`?  Seems like we
-        # could be missing out on a much more efficient/faster approach to this
-        # potentially large computation.
-        # state_seqs = np.broadcast_to(np.arange(M, dtype=int)[..., None], (M, N))
-        # log_lik_t = log_lik_fn(state_seqs)
-        log_lik_t = np.stack([log_lik_fn(np.broadcast_to(m, N)) for m in range(M)])
-
-        return ffbs_astep(gamma_0, Gammas_t, log_lik_t)
+        log_lik_t = log_lik_fn(point)
+
+        if set(self.lik_dict.keys()) == set([
+            "lik_n",
+            "alpha_n",
+            "beta_n",
+            "samples",
+            "alphas",
+        ]):
+            pass
+        else:
+            lik_n: np.ndarray = np.empty((M,), dtype=float)
+            alpha_n: np.ndarray = np.empty((M,), dtype=float)
+            beta_n: np.ndarray = np.empty((M,), dtype=float)
+            samples: np.ndarray = np.empty((N,), dtype=np.int8)
+            alphas: np.ndarray = np.empty((M, N), dtype=float)
```

Comment:
We need to do this initial setup in the constructor.

Also, we should pass these parameters directly to `ffbs_astep`, and not as a `dict` of parameters.